apps/api/src/app/workflows-v2/workflow.controller.ts

```diff
@@ -0,0 +1,118 @@
+import {
+  Body,
+  ClassSerializerInterceptor,
+  Controller,
+  Delete,
+  Get,
+  HttpStatus,
+  Logger,
+  Param,
+  Post,
+  Put,
+  Query,
+  Res,
+  UseGuards,
+  UseInterceptors,
+} from '@nestjs/common';
+import { Response } from 'express';
+import { ApiTags } from '@nestjs/swagger';
+import { DirectionEnum, UserSessionData } from '@novu/shared';
+import { ExternalApiAccessible, UserAuthGuard, UserSession } from '@novu/application-generic';
+import { ApiCommonResponses } from '../shared/framework/response.decorator';
+import { UserAuthentication } from '../shared/framework/swagger/api.key.security';
+import { GetWorkflowCommand } from './usecases/get-workflow/get-workflow.command';
+import { UpsertWorkflowUseCase } from './usecases/upsert-workflow/upsert-workflow.usecase';
+import { UpsertWorkflowCommand } from './usecases/upsert-workflow/upsert-workflow.command';
+import { GetWorkflowUseCase } from './usecases/get-workflow/get-workflow.usecase';
+import { ListWorkflowsUseCase } from './usecases/list-workflows/list-workflow.usecase';
+import { ListWorkflowsCommand } from './usecases/list-workflows/list-workflows.command';
+import { ListWorkflowResponse } from './dto/workflow-commons-fields';
+import { DeleteWorkflowUseCase } from './usecases/delete-workflow/delete-workflow.usecase';
+import { DeleteWorkflowCommand } from './usecases/delete-workflow/delete-workflow.command';
+import { GetListQueryParams } from './params/get-list-query-params';
+import { CreateWorkflowDto } from './dto/create-workflow-dto';
+import { UpdateWorkflowDto } from './dto/update-workflow-dto';
+import { WorkflowResponseDto } from './dto/workflow-response-dto';
+
+@ApiCommonResponses()
+@Controller({ path: `/workflows`, version: '2' })
+@UseInterceptors(ClassSerializerInterceptor)
+@UserAuthentication()
+@ApiTags('Workflows')
+export class WorkflowController {
+  constructor(
+    private upsertWorkflowUseCase: UpsertWorkflowUseCase,
+    private getWorkflowUseCase: GetWorkflowUseCase,
+    private listWorkflowsUseCase: ListWorkflowsUseCase,
+    private deleteWorkflowUsecase: DeleteWorkflowUseCase
+  ) {}
+
+  @Post('')
+  @UseGuards(UserAuthGuard)
+  create(
+    @UserSession() user: UserSessionData,
+    @Body() createWorkflowDto: CreateWorkflowDto
+  ): Promise<WorkflowResponseDto> {
+    return this.upsertWorkflowUseCase.execute(
+      UpsertWorkflowCommand.create({
+        workflowDto: createWorkflowDto,
+        user,
+      })
+    );
+  }
+  @Put(':workflowId')
```

Comment:
Please add a new line

Reply:
need to be handled on the eslint leve;l_